newsroom/template_filters.py

```diff
@@ -61,6 +70,84 @@ def parse_date(datetime):
     return datetime
 
 
+def get_schedule_type(start: datetime, end: datetime, all_day: bool, no_end_time: bool) -> ScheduleType:
+    duration = int((end - start).total_seconds() / 60)  # Convert seconds to minutes
+    DAY_IN_MINUTES = 24 * 60 - 1
+
+    if all_day:
+        return ScheduleType.ALL_DAY if duration == 0 else ScheduleType.MULTI_DAY
+
+    if no_end_time:
+        return ScheduleType.NO_DURATION
+
+    if duration > DAY_IN_MINUTES or start.date() != end.date():
+        return ScheduleType.MULTI_DAY
+
+    if duration == DAY_IN_MINUTES and start.date() == end.date():
+        return ScheduleType.ALL_DAY
+
+    if duration == 0 and start.time() == end.time():
+        return ScheduleType.NO_DURATION
+
+    return ScheduleType.REGULAR
+
+
+def is_item_tbc(item: dict) -> bool:
+    event_tbc = item.get("event", {}).get("_time_to_be_confirmed", False)
+    if not event_tbc:
+        planning = item.get("planning_items", [])
+        return planning[0].get("_time_to_be_confirmed", False) if planning else False
+    return event_tbc
+
+
+def format_event_datetime(item: dict) -> Optional[str]:
+    date_info = item.get("dates", {})
+
+    if not date_info:
+        return None
+
+    tz = date_info.get("tz", get_session_timezone())
+    try:
+        # Set the session timezone
+        set_session_timezone(tz)
+
+        start = parse_date(format_datetime(parse_date(date_info.get("start")), "yyyy-MM-dd HH:mm:ss"))
+        end = parse_date(format_datetime(parse_date(date_info.get("end")), "yyyy-MM-dd HH:mm:ss"))
+        all_day = date_info.get("all_day")
+        no_end_time = date_info.get("no_end_time")
+        is_tbc_item = is_item_tbc(item)
+
+        schedule_type = get_schedule_type(start, end, all_day, no_end_time)
+
+        formatted_start = datetime_long(start)
+        formatted_end = datetime_long(end)
+
+        if is_tbc_item:
```

Comment:
It seems like you're not covering the same scenarios as in the front-end. Front end has 9 different texts it can produce, this one only has 5 (see https://github.com/superdesk/newsroom-core/blob/develop/assets/agenda/utils.ts#L1140).

The front-end also has different outputs that contain the time only portion of the date, such as this one: https://github.com/superdesk/newsroom-core/blob/develop/assets/agenda/utils.ts#L1177 `{{startTime}} - {{endTime}} {{startDate}} {{timezone}}`